src/compiler/checker.ts

```diff
@@ -10136,13 +10137,37 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         })!.parent;
     }
 
+    function getInstanceTypeOfClassSymbol(classSymbol: Symbol): Type {
+        const classType = getDeclaredTypeOfSymbol(classSymbol) as GenericType;
+        const objectFlags = getObjectFlags(classType);
+        if (!(objectFlags & ObjectFlags.ClassOrInterface) || !classType.typeParameters) {
+            return classType;
+        }
+        const variances = getVariances(classType);
+        const typeArguments = map(classType.typeParameters, (typeParameter, i) => {
+            if (!strictInstanceOfTypeParameters) {
+                return anyType;
+            }
+            const variance = variances[i];
+            switch (variance & VarianceFlags.VarianceMask) {
+                case VarianceFlags.Independent:
+                case VarianceFlags.Bivariant:
+                    return anyType;
+                case VarianceFlags.Invariant:
+                    return unknownType;

```

Comment:
Thinking on it, `unknown` won't satisfy the constraint, so, lacking existentials,  `Invariant` should just be the constraint. This also seems to be missing the case where is makes a `VarianceFlags.Contravariant` into `never`.

Reply:
Will fix contravariant case. Although if contravariant type parameters are `never` then shouldn't invariant type parameters be `never` as well?

Also I'm unfamiliar with unique existential types, is that related to the "could be instantiated with a different subtype of constraint" check?

Reply:
>Also I'm unfamiliar with unique existential types, is that related to the "could be instantiated with a different subtype of constraint" check?

Yeah. We can't _perfectly_ represent what an unknown invariant type parameter's type should be (since that needs a type which is constrained to the type parameter's constraint but represents "some specific arbitrary type" - similar to a type parameter, but unscoped), so, lacking the ability to represent it perfectly, the constraint will at least be better for now. Contravariant uses will be too lax, but if it's `never`, then the _covariant_ uses will be too lax, and, on the balance, covariant uses are usually more common.

Reply:
`never` would allow you to do anything with covariant use sites, eg, your `() => T` would return a `never` that you could then fetch any member of (though we do prevent directly `.`'ing off a `never`, you can still use a lookup function ot the like, since `never`, in theory "has all members of all types").

Reply:
Ahh, I did not realize that behavior of `never`. Sounds good to me.